test/test_ff_utils.py

```diff
@@ -246,30 +356,68 @@ def test_unified_authentication(integrated_ff):
+    # This is ONLY an integration test. There is no unit test functionality tested here.
+    # All of the functionality used here is already tested elsewhere.
+
+    # Fourfront prod
+    ff_prod_auth = ff_utils.unified_authentication(ff_env="data")
+    assert len(ff_prod_auth) == 2
+    staging_auth = ff_utils.unified_authentication(ff_env="staging")
+    assert staging_auth == ff_prod_auth
+    green_auth = ff_utils.unified_authentication(ff_env="fourfront-green")
+    assert green_auth == ff_prod_auth
+    blue_auth = ff_utils.unified_authentication(ff_env="fourfront-blue")
+    assert blue_auth == ff_prod_auth
+
+    # CGAP prod
+    cgap_prod_auth = ff_utils.unified_authentication(ff_env="fourfront-cgap")
+    assert len(cgap_prod_auth) == 2
+
+    # Assure CGAP and Fourfront don't share auth
+    auth_is_shared = cgap_prod_auth == ff_prod_auth
+    assert not auth_is_shared
+
+    with raises_regexp(ClientError, "The specified bucket does not exist"):
+        # There is no such environment as 'fourfront-data'
         ff_utils.unified_authentication(ff_env="fourfront-data")
-    except ClientError as e:
-        assert "The specified bucket does not exist" in str(e)
-    else:
-        raise AssertionError("An exception was expected but did not occur.")
 
 
-@pytest.mark.integrated
-@pytest.mark.flaky
-def test_get_authentication_with_server(integrated_ff):
-    import copy
+def test_get_authentication_with_server_unit():
+
+    with mocked_s3_with_sse():
+
+        key1 = ff_utils.get_authentication_with_server(bar_env_auth_dict, None)
+        assert isinstance(key1, dict)
+        assert {'server', 'key', 'secret'} <= set(key1.keys())
+        assert key1['key'] == bar_env_auth_key
+        assert key1['secret'] == bar_env_auth_secret
+        assert key1['server'] == bar_env_url_trimmed
+
+        key2 = ff_utils.get_authentication_with_server(bar_env_default_auth_dict, None)
+        assert isinstance(key2, dict)
+        assert {'server', 'key', 'secret'} <= set(key2.keys())
+        assert key2 == key1
+
+        key3 = ff_utils.get_authentication_with_server(None, bar_env)
+        assert isinstance(key2, dict)
+        assert {'server', 'key', 'secret'} <= set(key3.keys())
+        assert key3 == key1
+
+        with raises_regexp(ValueError, 'ERROR GETTING SERVER'):
+            # The mocked foo_env, unlike the bar_env, is missing a 'server' key in its dict,
+            # so it will be rejected as bad data by .get_authentication_with_server().
+            ff_utils.get_authentication_with_server(foo_env_auth_dict, None)
+
+        with raises_regexp(ValueError, 'ERROR GETTING SERVER'):
+            # If the auth is not provided (or None), the ff_env must be given (and not None).
+            ff_utils.get_authentication_with_server(None, None)
+
+
+@pytest.mark.integratedx
+@pytest.mark.flaky  # TODO: Why is this marked flaky? There should be nothing flaky about this. -kmp 24-Oct-2020
```

Comment:
There are certain operations that can occur that can make this test fail - for example if a deployment or CNAME swap occurs during the run.

Reply:
Thanks. I removed the TODO and marked it this way, which I think we should probably do more often:
```
@pytest.mark.flaky  # e.g., if a deployment or CNAME swap occurs during the run
```